polyply/src/nanoparticle_constructor.py

```diff
@@ -0,0 +1,354 @@
+        # tag on surface of the core to attach with ligands
+        core_index_relevant = self.core_indices[
+            0
+        ]  # first entry of the identified core indices
+        # core_index_relevant_II = self.core_indices[
+        #    1
+        # ]  # second entry of the identified core indices
+        attachment_list = []
+        # find the indices that are equivalent to the atom
+        # on the ligand we wish to attach onto the NP
+        # based on the ligands_N, and then merge onto the
+        # NP core the main ligand blocks ligand_N number of times
+
+        for index in range(0, self.ligands_N):
+            ligand_index = index * ligand_len  # compute starting index for each ligand
+            print(ligand_index, core_size)
+            scaled_ligand_index = ligand_index + core_size
+            attachment_list.append(scaled_ligand_index)
+            # merge the relevant molecule
+            self.np_molecule.merge_molecule(self.ligand_block)
+
+        # create list of interactions for first batch of indices we have on the
+        # core surface
+        for index, index2 in enumerate(
+            list(core_index_relevant.keys())[: self.ligands_N]
+        ):
+            interaction = vermouth.molecule.Interaction(
+                atoms=(index2, attachment_list[index] + attachment_index),
+                parameters=["1", "0.0033", "50000"],
+                meta={},
+            )
+            # append onto bonds the interactions between equivalent indices
+            self.np_molecule.interactions["bonds"].append(interaction)
+
+        # create list for interactions for second batch of indices we have on the core surfaces
+        # for index, index2 in enumerate(
+        #    list(core_index_relevant_II.keys())[: self.ligands_N]
+        # ):
+        #    interaction = vermouth.molecule.Interaction(
+        #        atoms=(index2, atom_max + (index * length) + attachment_index),
+        #        parameters=["1", "0.0033", "50000"],
+        #        meta={},
+        #    )
+        #    # append onto bonds the interactions between equivalent indices
+        #    self.np_molecule.interactions["bonds"].append(interaction)
+
+        for node in self.np_molecule.nodes:
+            # change resname to moltype
+            self.np_molecule.nodes[node]["resname"] = "TEST"
+
+        self.graph = MetaMolecule._block_graph_to_res_graph(
+            self.np_molecule
+        )  # generate residue graph
+
+        # generate meta molecule from the residue graph with a new molecule name
+        self.meta_mol = MetaMolecule(
+            self.graph, force_field=self.force_field, mol_name="test"
+        )
+
+        self.np_molecule.meta["moltype"] = "TEST"
+        # reassign the molecule with the np_molecule we have defined new interactions with
+        self.meta_mol.molecule = self.np_molecule
+        return self.meta_mol
+
+    def create_itp(self, write_path):
```

Comment:
this function is not needed. Processors just return molecules. I/O is handled at the level of gen_np